### PR TITLE
chore: Golf simp calls in Data directory

### PR DESCRIPTION
**Pull request by BoltonBailey**
This PR is a spiritual successor to [this](https://github.com/leanprover-community/mathlib/pull/17062) mathlib3 one, which sought to remove unnecessary lemmas from `simp` calls, and convert simp calls to `simp_rw` for readability. It does this with a Python script. This PR focuses on removing lemmas, #9101 looks to add more features.

---

- [x] depends on: #9095 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by BoltonBailey** [comment]
Come to think of it, I don't see a particular reason this only has to work on `simp only` it could be made to do the same for `rw`/`simp [...]`/ tactics in general.

File: Mathlib/Data/DFinsupp/Basic.lean [hunk 1]
```diff
@@ -166,7 +166,7 @@ theorem mapRange_comp (f : ∀ i, β₁ i → β₂ i) (f₂ : ∀ i, β i → 
     (hf₂ : ∀ i, f₂ i 0 = 0) (h : ∀ i, (f i ∘ f₂ i) 0 = 0) (g : Π₀ i : ι, β i) :
     mapRange (fun i => f i ∘ f₂ i) h g = mapRange f hf (mapRange f₂ hf₂ g) := by
   ext
-  simp only [mapRange_apply]; rfl
+  simp only []; rfl
```
Comment (alreadydone) [inline Mathlib/Data/DFinsupp/Basic.lean:169]: I suspect all `simp only []` could be replaced by `dsimp only` (or at least `simp only`)

Comment (BoltonBailey) [reply Mathlib/Data/DFinsupp/Basic.lean:169]: In this case it seems that just `rfl` works.

I'm starting to get ambitious about this. Why not write a whole bunch of golfers like the ones I already have: One to remove lemmas from simps, one to convert `simp only []` to `dsimp only`, one to convert `simp only` to `simp_rw`, and many more?

**Comment by BoltonBailey** [comment]
Ok I think perhaps we can just let this PR remove simp lemmas, I made #9101 to do other things.

**Comment by BoltonBailey** [comment]
!bench

**Comment by BoltonBailey** [comment]
See [this comment](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.239079.20Golfing.20simps/near/408359275)

**Comment by BoltonBailey** [comment]
Closing in favor of #9320

**Pull request closed without merging**